api/dataset.go

```diff
@@ -220,6 +220,19 @@ func (api *DatasetAPI) getVersions(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var hasInvalidState bool
+	for _, item := range results.Items {
+		if err = models.CheckState("version", item.State); err != nil {
+			hasInvalidState = true
+			log.ErrorC("unpublished version has an invalid state", err, log.Data{"state": item.State})
+		}
+	}
+
+	if hasInvalidState {
```

Comment:
Can we log a message here?

Reply:
At least 1 log message would have already occurred in the for loop above (in some cases multiple times but that is so we are able to fix the data for each document that has an invalid status); so I don't feel the need to have another log here.